src/traits/networking/centralized_web_server_network.rs

```diff
@@ -0,0 +1,720 @@
+                    let deserialized_message = bincode::deserialize(message).unwrap();
+                    deserialized_messages.push(deserialized_message);
+                }
+                Ok(Some(deserialized_messages))
+            }
+            Ok(None) => Ok(None),
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+#[serde(bound(deserialize = ""))]
+/// A message being sent to the web server
+pub struct SendMsg<M: NetworkMsg> {
+    /// The optional message, or body, to send
+    message: Option<M>,
+    /// The endpoint to send the message to
+    endpoint: String,
+}
+
+/// A message being received from the web server
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+#[serde(bound(deserialize = ""))]
+pub struct RecvMsg<M: NetworkMsg> {
+    /// The optional message being received
+    message: Option<M>,
+}
+
+/// Trait for messages being sent to the web server
+pub trait SendMsgTrait<M: NetworkMsg> {
+    /// Returns the endpoint to send the message to
+    fn get_endpoint(&self) -> String;
+    /// Returns the actual message being sent
+    fn get_message(&self) -> Option<M>;
+}
+
+/// Trait for messages being received from the web server
+pub trait RecvMsgTrait<M: NetworkMsg> {
+    /// Returns the actual message being received
+    fn get_message(&self) -> Option<M>;
+}
+
+impl<M: NetworkMsg> SendMsgTrait<M> for SendMsg<M> {
+    fn get_endpoint(&self) -> String {
+        self.endpoint.clone()
+    }
+
+    fn get_message(&self) -> Option<M> {
+        self.message.clone()
+    }
+}
+
+impl<M: NetworkMsg> RecvMsgTrait<M> for RecvMsg<M> {
+    fn get_message(&self) -> Option<M> {
+        self.message.clone()
+    }
+}
+
+impl<M: NetworkMsg> NetworkMsg for SendMsg<M> {}
+impl<M: NetworkMsg> NetworkMsg for RecvMsg<M> {}
+
+impl<
+        M: NetworkMsg + 'static,
+        K: SignatureKey + 'static,
+        E: ElectionConfig + 'static,
+        TYPES: NodeType + 'static,
+        PROPOSAL: ProposalType<NodeType = TYPES> + 'static,
+        VOTE: VoteType<TYPES> + 'static,
+    > CentralizedWebServerNetwork<M, K, E, TYPES, PROPOSAL, VOTE>
+{
+    #[allow(clippy::panic)]
+    /// Creates a new instance of the `CentralizedWebServerNetwork`
+    /// # Panics
+    /// if the web server url is malformed
+    pub fn create(
+        host: &str,
+        port: u16,
+        wait_between_polls: Duration,
+        key: TYPES::SignatureKey,
+    ) -> Self {
+        let base_url_string = format!("http://{host}:{port}");
+        error!("Connecting to web server at {base_url_string:?}");
+
+        let base_url = base_url_string.parse();
+        if base_url.is_err() {
+            error!("Web server url {:?} is malformed", base_url_string);
+        }
+
+        // TODO ED Wait for healthcheck
+        let client = surf_disco::Client::<ClientError>::new(base_url.unwrap());
+
+        let inner = Arc::new(Inner {
+            phantom: PhantomData::default(),
```

Comment:
nit: I think we can even do
```suggestion
            phantom: PhantomData,
```